crates/api/src/constraints/api.rs

```diff
@@ -108,14 +134,18 @@ where
             return Err(ConstraintsApiError::InvalidConstraints)
         }
 
-        // PERF: can we avoid calling the db?
-        let maybe_validator_pubkey = api.db.get_proposer_duties().await?.iter().find_map(|d| {
-            if d.slot == first_constraints.slot {
-                Some(d.entry.registration.message.public_key.clone())
+        let maybe_validator_pubkey =
+            if let Some(duties) = api.curr_slot_info.read().await.new_duties.as_ref() {
```

Comment:
are you sure we need the `tokio::sync` version of RwLock? It doesn't look like we're holding the lock through an await point. If `parking_lot` is available, I'd use that, and if not, perhaps the `std` lib one should be fine too

Reply:
You're right -- for the reason you mentioned and also because there is not much contention over the resource, `std::sync::RwLock` is probably faster. However in all the codebase `tokio::sync::RwLock` is used, so for consistency I'll keep it as it is.